ostd/src/arch/x86/dispatcher.rs

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::{borrow::ToOwned, vec::Vec};
+
+use align_ext::AlignExt;
+
+use crate::{
+    boot::memory_region::MemoryRegionType, device::dispatcher::io_mem::IoMemDispatcherBuilder,
+};
+
+/// Initializes the allocatable MMIO area based on the x86-64 memory distribution map.
+///
+/// In x86-64, the available physical memory area is divided into two regions below 32 bits (Low memory)
+/// and above (High memory). Where the area from the top of low memory to 0xffff_ffff, after the top of
+/// high memory, is the available MMIO area.
+pub(super) fn construct_io_mem_dispatcher_builder() -> IoMemDispatcherBuilder {
+    let mut regions = crate::boot::memory_regions().to_owned();
+    assert!(!regions.is_empty());
+    regions.sort_by_key(|a| a.base());
+    regions.reverse();
```

Comment:
```suggestion
    let reversed_regions = {
        let mut regions = crate::boot::memory_regions().to_owned();
        assert!(!regions.is_empty());
        regions.sort_by_key(|a| a.base());
        regions.reverse();
        regions
   };
```

This version also seems to be a bit better. I think `reversed` is important information that we want to emphasize a bit in the variable name to make the following logic easier to understand.